### PR TITLE
EDU-793 Transform rooms domain model to keep list of included documents

### PR DESCRIPTION
**Pull request by adrianaluchian**
Closes https://educandu.atlassian.net/browse/EDU-793

### CODE REVIEW

**Comment by adrianaluchian** [comment]
One thing missing: there are no tests for `hardDeleteDocument`, so I will include them soon in this PR.

File: src/services/document-service.js [hunk 1]
```diff
@@ -175,6 +176,12 @@ class DocumentService {
 
         await this.documentRevisionStore.saveDocumentRevision(newRevision, { session });
         await this.documentStore.saveDocument(newDocument, { session });
+
+        if (newDocument.roomId) {
```
Comment (ahelmberger) [inline src/services/document-service.js:180]: If the `roomId` is set, we should have also taken the room lock at the beginning of this `try` block, so no other concurrent room updates can take place.

File: src/services/document-service.js [hunk 2]
```diff
@@ -255,9 +262,17 @@ class DocumentService {
     try {
       lock = await this.lockStore.takeDocumentLock(documentId);
 
-      logger.info(`Hard deleting external document '${documentId}'`);
+      logger.info(`Hard deleting document '${documentId}'`);
 
       await this.transactionRunner.run(async session => {
+        const doc = await this.documentStore.getDocumentById(documentId, { session });
+
+        if (doc.roomId) {
```
Comment (ahelmberger) [inline src/services/document-service.js:270]: I'd also lock the room for this operation

File: src/services/document-service.js (hunk 1 above)
Comment (adrianaluchian) [reply src/services/document-service.js:184]: Good point, thanks! Took the lock.

File: src/services/document-service.js (hunk 2 above)
Comment (adrianaluchian) [reply src/services/document-service.js:279]: Done here too.
And changed the logic for updating rooms as we discussed 🙏

Comment (ahelmberger) [reply src/services/document-service.js:279]: Nice! Looks good, thanks!

**Review by ahelmberger — APPROVED** [review]

**Pull request merged**